src/commands/dev-env-import-sql.ts

```diff
@@ -111,38 +120,44 @@ export class DevEnvImportSQLCommand {
 			fs.unlinkSync( resolvedPath );
 		}
 
-		const cacheArg = [ 'wp', 'cache', 'flush', '--skip-plugins', '--skip-themes' ].concat(
-			this.options.quiet ? '--quiet' : []
-		);
-		await exec( lando, this.slug, cacheArg );
+		await flushCache( lando, this.slug, this.options.quiet );
 
 		if (
 			undefined === this.options.skipReindex ||
 			! processBooleanOption( this.options.skipReindex )
 		) {
 			try {
-				await exec( lando, this.slug, [ 'wp', 'cli', 'has-command', 'vip-search' ] );
-				await exec( lando, this.slug, [
-					'wp',
-					'vip-search',
-					'index',
-					'--setup',
-					'--network-wide',
-					'--skip-confirm',
-				] );
+				await reIndexSearch( lando, this.slug );
 			} catch {
 				// Exception means they don't have vip-search enabled.
 			}
 		}
 
-		const addUserArg = [
-			'wp',
-			'dev-env-add-admin',
-			'--username=vipgo',
-			'--password=password',
-			'--skip-plugins',
-			'--skip-themes',
-		].concat( this.options.quiet ? '--quiet' : [] );
-		await exec( lando, this.slug, addUserArg );
+		await addAdminUser( lando, this.slug );
+	}
+
+	public getImportArgs( dumpDetails: SqlDumpDetails ) {
+		let importArg = [ 'db', '--disable-auto-rehash' ].concat(
+			this.options.quiet ? '--silent' : []
+		);
+		const threadCount = Math.max( os.cpus().length - 2, 1 );
+		if ( dumpDetails.type === SqlDumpType.MYDUMPER ) {
+			importArg = [
+				'db-myloader',
+				'--overwrite-tables',
+				`--source-db=${ dumpDetails.sourceDb }`,
+				`--threads=${ threadCount }`,
+				'--max-threads-for-schema-creation=10',
+				'--max-threads-for-index-creation=10',
+				'--skip-triggers',
+				'--skip-post',
+				'--innodb-optimize-keys',
+				'--checksum=SKIP',
+				'--metadata-refresh-interval=2000000',
+				'--stream',
+			].concat( this.options.quiet ? [ '--verbose=0' ] : [ '--verbose=3' ] );
+		}
+
+		return importArg;
```

Comment:
That's a nice refactor! I bet it would be easier to review if you package just the refactor 👍

Reply:
I'll separate it out then :)

Reply:
Please tag me there if you can 🙏

Reply:
#1962